### PR TITLE
fix(querylog): Flush querylog producer before terminating processes

### PR DESCRIPTION
**Pull request by lynnagara**
Another attempt to fix that producer termination with unflushed events error.
During the earlier test we discovered there were no unflushed events on the
subscription results producer so that one is unlikely to be the actual
culprit here. This time we are calling flush on the querylog producer.

### CODE REVIEW

File: snuba/state/__init__.py [hunk 1]
```diff
@@ -325,6 +325,12 @@ def record_query(query_metadata: Mapping[str, Any]) -> None:
         logger.exception("Could not record query due to error: %r", ex)
 
 
+def flush_producer() -> None:
+    global kfk
+    if kfk is not None:
+        kfk.flush()
```
Comment (MeredithAnya) [inline snuba/state/__init__.py:331]: do we not want to give this a timeout?

Comment (lynnagara) [reply snuba/state/__init__.py:331]: What timeout do you suggest? In arroyo producer we don't so I think it's fine https://github.com/getsentry/arroyo/blob/4854c67b5ad19c304e3301eff98c45ccaed93a5e/arroyo/backends/kafka/consumer.py#L681

**Review by evanh — APPROVED** [review]
Looks good to me, pending Meredith's question.

**Comment by lynnagara** [comment]
> Looks good to me, pending Meredith's question.

Going to merge this now. Confirmed this offline with Meredith.

**Pull request merged**